src/object/srv_obj_migrate.c

```diff
@@ -452,7 +458,7 @@ migrate_pool_tls_create_one(void *data)
 	pool_tls->mpt_executed_ult = 0;
 	pool_tls->mpt_root_hdl = DAOS_HDL_INVAL;
 	pool_tls->mpt_max_eph = arg->max_eph;
-	pool_tls->mpt_pool = ds_pool_child_lookup(arg->pool_uuid);
+	pool_tls->mpt_pool = child;
```

Comment:
ds_pool_child could leak when any of above operation fails. I think simpler change could be:

`
pool_tls->mpt_pool = ds_pool_child_lookup(arg->pool_uuid);
 if (pool_tls->mpt_pool == NULL)
    D_GOTO(out, rc = -DER_NO_HDL);
`